### DISCUSSION TITLE
Обращение к данным временной таблицы из временной таблицы

### DISCUSSION THREAD

**Discussion by an220022** [totumonline/totum-ru-issues-and-discussions#9] [Как это сделать?] [answered]
Как можно обратиться к данным временной таблицы из временной таблицы?

Т.е. командой из одной таблицы сформировать временную таблицу с некоторыми параметрами (массив данных), а потом на основе этих данных заполнить h_insert для заполнения строк таблицы

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Добрый день — это классная задача, которая решается при помощи параметра `hide: true` функции `linkToDataTable`. Если этого параметра нет, то `linkToDataTable` открывает окно, а если есть `hide: true` — то окно не открывается, а функция вернет назад `hash` созданной временной таблицы.

Те получается так:

+ Вызывается временная таблица, которая должна заполниться по `insert`

+ `h_insert` взывает еще одну временную с параметром `hide: true`

+ Вложенная временная делает нужное вычисление

+ `h_insert` получив `hash` вложенной временной забирает из нее данные

Все это упаковывается в код `h_insert` примерно так:

```totum

=: selectList(table: 'tmp_table_2'; hash: $linkto; field: 'id')

linkto: linkToDataTable(table: 'tmp_table_2'; params: $row; hide: true)

row: rowCreate(....)

``` 

Вот еще ссылка на доки: https://docs.totum.online/specific-codes#linktodatatable
